File: lib/models/reward_redemption.dart
class RewardRedemption {
  final int id;
  final int productId;
  final String? productName;
  final int quantity;
  final int pointsTotal;
  final String status;
  final String statusLabel;
  final String? trackingNumber;
  final String createdAt;

  const RewardRedemption({
    required this.id,
    required this.productId,
    this.productName,
    required this.quantity,
    required this.pointsTotal,
    required this.status,
    required this.statusLabel,
    this.trackingNumber,
    required this.createdAt,
  });

  factory RewardRedemption.fromJson(Map<String, dynamic> json) {
    return RewardRedemption(
      id: json['id'] as int,
      productId: (json['product_id'] as num?)?.toInt() ?? 0,
      productName: json['product_name'] as String?,
      quantity: (json['quantity'] as num?)?.toInt() ?? 0,
      pointsTotal: (json['points_total'] as num?)?.toInt() ?? 0,
      status: json['status'] as String? ?? 'pending',
      statusLabel: json['status_label'] as String? ?? '',
      trackingNumber: json['tracking_number'] as String?,
      createdAt: json['created_at'] as String? ?? '',
    );
  }
}

File: clinic-backend/app/Models/RewardRedemption.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class RewardRedemption extends Model
{
    use HasFactory;

    protected $fillable = [
        'user_id',
        'product_id',
        'quantity',
        'points_per_item',
        'points_total',
        'status',
        'shipping_address_id',
        'tracking_number',
        'notes',
        'admin_notes',
        'approved_at',
        'shipped_at',
        'delivered_at',
    ];

    protected $casts = [
        'quantity' => 'integer',
        'points_per_item' => 'integer',
        'points_total' => 'integer',
        'approved_at' => 'datetime',
        'shipped_at' => 'datetime',
        'delivered_at' => 'datetime',
    ];

    const STATUS_PENDING = 'pending';

    const STATUS_APPROVED = 'approved';

    const STATUS_PREPARING = 'preparing';

    const STATUS_SHIPPED = 'shipped';

    const STATUS_DELIVERED = 'delivered';

    const STATUS_CANCELLED = 'cancelled';

    public function user()
    {
        return $this->belongsTo(User::class);
    }

    public function product()
    {
        return $this->belongsTo(Product::class);
    }

    public function shippingAddress()
    {
        return $this->belongsTo(CustomerAddress::class, 'shipping_address_id');
    }

    public function scopePending($query)
    {
        return $query->where('status', self::STATUS_PENDING);
    }

    public function scopeActive($query)
    {
        return $query->whereNotIn('status', [self::STATUS_CANCELLED, self::STATUS_DELIVERED]);
    }

    public function getStatusLabelAttribute()
    {
        return match ($this->status) {
            self::STATUS_PENDING => 'รอดำเนินการ',
            self::STATUS_APPROVED => 'อนุมัติแล้ว',
            self::STATUS_PREPARING => 'กำลังจัดเตรียม',
            self::STATUS_SHIPPED => 'จัดส่งแล้ว',
            self::STATUS_DELIVERED => 'ส่งถึงแล้ว',
            self::STATUS_CANCELLED => 'ยกเลิก',
            default => $this->status,
        };
    }
}
